diag_manager/fms_diag_yaml.F90

```diff
@@ -42,12 +44,28 @@ module fms_diag_yaml_mod
 public :: diag_yaml_object_init, diag_yaml_object_end
 public :: diagYamlObject_type, get_diag_yaml_obj, get_title, get_basedate, get_diag_files, get_diag_fields
 public :: diagYamlFiles_type, diagYamlFilesVar_type
+public :: get_num_unique_fields
+
 !> @}
 
 integer, parameter :: basedate_size = 6
 integer, parameter :: NUM_SUB_REGION_ARRAY = 8
 integer, parameter :: MAX_STR_LEN = 255
 
+!> @brief type to hold an array of sorted diag_fiels
+type varList
+  character(len=255), allocatable :: var_name(:) !< Array of diag_field
+  type(c_ptr), allocatable :: var_pointer(:) !< Array of pointers
+  integer, allocatable :: diag_field_indices(:) !< Index of the field in the diag_field array
+end type
+
+!> @brief type to hold an array of sorted diag_files
+type fileList
```

Comment:
this should be fileList_type

Reply:
done in https://github.com/NOAA-GFDL/FMS/pull/944/commits/13611692fb9996116f9d871b6a81c5b528d095fc